jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/MessageHandler.java

```diff
@@ -86,48 +85,14 @@ protected void onBinary(ByteBuffer message, Callback callback)
         };
     }
 
-    private Logger LOG = Log.getLogger(MessageHandler.class);
-
-    private final int factor;
+    protected static final Logger LOG = Log.getLogger(MessageHandler.class);
+    private static final int BUFFER_SIZE = 65535;
 
     private CoreSession coreSession;
-    private Utf8StringBuilder utf8StringBuilder = null;
-    private ByteBuffer binaryMessage = null;
+    private Utf8StringBuilder textMessageBuffer = new Utf8StringBuilder(BUFFER_SIZE);
+    private ByteArrayOutputStream binaryMessageBuffer = new ByteArrayOutputStream(BUFFER_SIZE);
```

Comment:
I like the lazy instantiation of these because many handlers will only and text OR binary and rarely both.  Initialising both here means you are likely to grab a 64K buffer that is probably unused!

Also, why hard code a 64K buffer size?  It should at least be based of maxMessageSize... but I think it is OK to start with a small capacity and let them grow to what is needed.  A lot of ws messages are very small, so it is a waste to allocate a 64K buffer for them and worse to allocated two!!!

Reply:
I was using the hard coded buffer size because that's what was being done in the `MessageSinks`.
https://github.com/eclipse/jetty.project/blob/35dbce4cce2bca4be6ca57b58be80bdbb62ad868/jetty-websocket/jetty-websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/ByteArrayMessageSink.java#L36-L45

Reply:
I am now also reusing the `Utf8StringBuilder` and `ByteArrayOutputStream` and just calling `reset()` instead of creating a new one for every message.